src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -1108,6 +1194,7 @@ def __new__(
         electrical_series_name: str | None = None,
         sampling_frequency: float | None = None,
         samples_for_rate_estimation: int = 1000,
+        unit_table_name: str = "units",
```

Comment:
@h-mayorquin actually that is default already (note that this is the main `NwbSorting`) class

Reply:
Yeah, but you still call the `retrieve_unit_table` which actually does two things (my fault). Performs a search which is the costly part and then fetches.  I am saying we should avoid calling this function as the search is the costly part.


https://github.com/alejoe91/spikeinterface/blob/33a7babfe03cc5c1a4df3ff3434b72c660558d86/src/spikeinterface/extractors/nwbextractors.py#L75-L109

Reply:
Got it!